answerking_app/utils/model_types.py

```diff
@@ -1,42 +1,63 @@
-from typing import Any
+import datetime
 
 from typing_extensions import (  # for Python <3.11 with (Not)Required
     NotRequired,
     TypedDict,
 )
+from typing import Any
 
 
-class ItemType(TypedDict):
+class ProductType(TypedDict):
     id: NotRequired[int]
     name: str
-    price: str
+    price: str | float
     description: str
+    categories: NotRequired["list[CategoryType]"]
     retired: NotRequired[bool]
-    stock: int
-    calories: int
+
+
+class CategoryProductType(TypedDict):
+    id: int
 
 
 class CategoryType(TypedDict):
     id: NotRequired[int]
     name: NotRequired[str]
+    description: NotRequired[str]
+    createdOn: NotRequired[datetime.datetime | str]
+    lastUpdated: NotRequired[datetime.datetime | str]
```

Comment:
are the dates not always strings?

Reply:
for some of the tests the way we were doing them it had to be datetime

Reply:
ok thats fine. should we not be setting the type to datetime.datetime | str everywhere then?